crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -115,7 +114,7 @@ public Mediator(Metadata md,
      * implementation can perform further validations.
      */
     @StopWatch(loggerName = "stopwatch.com.redhat.lightblue.mediator.Mediator", sizeCalculatorClass = "com.redhat.lightblue.mediator.ResponsePayloadSizeCalculator")
-    public Response insert(InsertionRequest req) {
+    public Response insert(InsertionRequest req, RequestMetrics.Context metricCtx) {
```

Comment:
I don't like metrics context being passed like this. I can think of a few options that might work better.

One option is to use OperationContext to keep RequestMetrics.Context. But, operation context is constructed in mediator calls (newCtx), so extending mediator with a new newCtx method that sets the metrics could work.

Another method is to use a ThreadLocal in RequetMetrics.Context to set it in rest layer, and get it in newCtx() in mediator.

The code needs to handle null request metrics. It can be done by using a noop metrics implementation.